buildSrc/src/main/java/org/opensearch/gradle/BwcVersions.java

```diff
@@ -342,7 +325,7 @@ private Map<Integer, List<Version>> getReleasedMajorGroupedByMinor() {
 
     public void compareToAuthoritative(List<Version> authoritativeReleasedVersions) {
         Set<Version> notReallyReleased = new HashSet<>(getReleased());
-        notReallyReleased.removeAll(authoritativeReleasedVersions);
+        authoritativeReleasedVersions.forEach(notReallyReleased::remove);
```

Comment:
Just curious, why `removeAll` didn't cut?

Reply:
The performance of `set.removeAll` depends upon the size of the set vs the passed collection and the performance of the `contains` method of the passed collection. [Here is the Java Doc](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/AbstractSet.html#removeAll(java.util.Collection)). In this case, the method is called  by passing the list of released versions from [maven](https://repo1.maven.org/maven2/org/opensearch/opensearch/maven-metadata.xml) which is going to be bigger than the list of released versions in [Versions.java](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/Version.java#L78-L88)

Reply:
That is correct, and the `removeAll` is smart enough to make the decision (according to same Javadoc), if we really care about performance here (which is always good but may be sometimes not strictly necessary),  we probably should represent the `List<Version> authoritativeReleasedVersions` as a `Set<Version>` (why do we need duplicates here?) in the first place, wdyt?

Reply:
Yes, we could change the type to set but it would require some further changes. Do you have any particular concern with this refactoring?

Reply:
Not, not necessarily the concern but curiosity